shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.mm

```diff
@@ -129,6 +131,64 @@ - (void)clipRRect:(const SkRRect&)clipSkRRect {
   CGPathRelease(pathRef);
 }
 
+- (void)clipPath:(const SkPath&)path {
+  CGMutablePathRef pathRef = CGPathCreateMutable();
+  if (!path.isValid()) {
+    return;
+  }
+  if (path.isEmpty()) {
+    CAShapeLayer* clip = [[CAShapeLayer alloc] init];
+    clip.path = pathRef;
+    self.layer.mask = clip;
+    CGPathRelease(pathRef);
+    return;
+  }
+
+  // Loop through all verbs and translate them into CGPath
+  SkPath::Iter iter(path, true);
+  SkPoint pts[kMaxPointsInVerb];
+  SkPath::Verb verb = iter.next(pts);
+  while (verb != SkPath::kDone_Verb) {
+    switch (verb) {
+      case SkPath::kMove_Verb: {
+        CGPathMoveToPoint(pathRef, nil, pts[0].x(), pts[0].y());
+        break;
+      }
+      case SkPath::kLine_Verb: {
+        CGPathAddLineToPoint(pathRef, nil, pts[1].x(), pts[1].y());
```

Comment:
are we sure we can assume that pts[0] is the last point of the previous verb?

Reply:
Offline discussion: We confirmed we can assume the pts[0] is always the last point of the previous verb, but we are going to add a DCHECK for our own sanity.